src/app/gps/qgsgpsinformationwidget.cpp

```diff
@@ -304,8 +358,8 @@ QgsGpsInformationWidget::~QgsGpsInformationWidget()
     disconnectGps();
   }
 
-  delete mpMapMarker;
-  delete mpRubberBand;
+  delete mMapMarker;
```

Comment:
If it inherits QObject, yes.
I think markers are map canvas items (QGraphcisItem) which don't inherit QObject, so this won't work. It would be good to come up with a similar concept for map canvas items, but I from a quick look it's not trivial. As long as it's guaranteed that this is deleted earlier than the map canvas, it's safe to do.